tests/explainers/test_tree.py

```diff
@@ -145,6 +145,26 @@ def test_xgboost_multiclass():
     # ensure plot works for first class
     shap.dependence_plot(0, shap_values[0], X, show=False)
 
+def test_xgboost_ranking():
+    try:
+        import xgboost
+    except:
+        print("Skipping test_xgboost_ranking!")
+        return
+    import shap
+
+    # train lightgbm ranker model
+    x_train, y_train, x_test, y_test, q_train, q_test = shap.datasets.rank()
+    params = {'objective': 'rank:pairwise', 'learning_rate': 0.1,
+              'gamma': 1.0, 'min_child_weight': 0.1,
+              'max_depth': 4, 'n_estimators': 4}
+    model = xgboost.sklearn.XGBRanker(**params)
+    model.fit(x_train, y_train, q_train.astype(int),
+              eval_set=[(x_test, y_test)], eval_group=[q_test.astype(int)])
+
+    # explain the model's predictions using SHAP values
+    shap.TreeExplainer(model).shap_values(x_test)
```

Comment:
Could you also compare the sum of these values to the margin prediction of the model? Otherwise all we are doing is checking that the code runs.

Reply:
good idea, done!